start.php

```diff
@@ -1,145 +1,145 @@
+	elgg_register_plugin_hook_handler('prepare', 'notification:publish:object:scheduling_poll', '\Elgg\Scheduling\Notification::prepare');
+	elgg_register_plugin_hook_handler('prepare', 'notification:update:object:scheduling_poll', '\Elgg\Scheduling\Notification::prepare');
+	elgg_register_plugin_hook_handler('get', 'subscriptions', '\Elgg\Scheduling\Notification::subscribers');
+
+	add_group_tool_option('scheduling', elgg_echo('scheduling:group:enable'));
+
+	elgg_extend_view('css/elgg', 'scheduling/css');
+
+	elgg_register_menu_item('site', array(
+		'name' => 'scheduling',
+		'text' => elgg_echo('scheduling'),
+		'href' => 'scheduling',
+	));
+        
+        // Register entity type for search
+	elgg_register_entity_type('object', 'scheduling_poll');
+}
+
+/**
+ * Handle calls to page "scheduling"
+ *
+ * @param array $page
+ */
+function scheduling_page_handler($page) {
+	elgg_load_library('scheduling');
+
+	if (!isset($page[0])) {
+		$page[0] = 'all';
+	}
+
+	elgg_push_breadcrumb(elgg_echo('scheduling'));
+
+	$base_path = elgg_get_plugins_path() . 'scheduling/pages/';
+
+	switch ($page[0]) {
+		case 'add':
+			set_input('container_guid', $page[1]);
+			$page_path = 'save.php';
+			break;
+		case 'edit':
+			set_input('guid', $page[1]);
+			$page_path = 'save.php';
+			break;
+		case 'view':
+			set_input('guid', $page[1]);
+			$page_path = 'view.php';
+			break;
+		case 'days':
+			set_input('guid', $page[1]);
+			$page_path = 'days.php';
+			break;
+		case 'owner';
+			$user = get_user_by_username($page[1]);
+			if ($user) {
+				set_input('container_guid', $user->guid);
+			}
+			$page_path = 'all.php';
+			break;
+		case 'group':
+			if ($page[1]) {
+				set_input('container_guid', $page[1]);
+			}
+			$page_path = 'all.php';
+			break;
+		default:
+			$page_path = 'all.php';
+			break;
+	}
+
+	include_once($base_path . $page_path);
+
+	return true;
+}
+
+/**
+ * Populates the ->getUrl() method
+ *
+ * @param string $hook
+ * @param string $type
+ * @param string $url
+ * @param array  $params
+ * @return string URL
+ */
+function scheduling_poll_url($hook, $type, $url, $params) {
+	$entity = $params['entity'];
+
+	if (!$entity instanceof ElggSchedulingPoll) {
+		return $url;
+	}
+
+	$title = elgg_get_friendly_title($entity->name);
+
+	return "scheduling/view/{$entity->guid}/$title";
+}
+
+/**
+ * Set up entity menu items
+ *
+ * @param string         $hook
+ * @param string         $type
+ * @param array          $menu
+ * @param ElggMenuItem[] $params
+ * @return ElggMenuItem[] $menu
+ */
+function scheduling_entity_menu($hook, $type, $menu, $params) {
+	if (elgg_extract('handler', $params) !== 'scheduling') {
+		return $menu;
+	}
+
+	$entity = $params['entity'];
+
+	if ($entity->canEdit()) {
+		$menu[] = ElggMenuItem::factory(array(
+			'name' => 'scheduling_time',
+			'href' => "scheduling/days/{$entity->guid}",
+			'text' => elgg_echo('scheduling:edit:time'),
+		));
+	}
+
+	return $menu;
+}

```

Comment:
What happened to this file? It seems like every line got changed.

Reply:
Sorry, my editor had changed the end of line chars